controller/sync_strats/sync_instant.go

```diff
@@ -0,0 +1,528 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package sync_strats
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/lucsky/cuid"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/edge/build"
+	"github.com/openziti/edge/controller/env"
+	"github.com/openziti/edge/controller/handler_edge_ctrl"
+	"github.com/openziti/edge/controller/model"
+	"github.com/openziti/edge/controller/persistence"
+	"github.com/openziti/edge/pb/edge_ctrl_pb"
+	"github.com/openziti/fabric/controller/network"
+	"github.com/openziti/foundation/channel2"
+	"github.com/openziti/foundation/storage/ast"
+	"go.etcd.io/bbolt"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	RouterSyncStrategyInstant env.RouterSyncStrategyType = "instant"
+)
+
+var _ env.RouterSyncStrategy = &InstantStrategy{}
+
+// Original API Session synchronization implementation. Assumes that on connect, the router requires and instant
+// and full set of API Sessions
+type InstantStrategy struct {
+	InstantStrategyOptions
+
+	rtxMap *routerTxMap
+
+	helloHandler  channel2.ReceiveHandler
+	resyncHandler channel2.ReceiveHandler
+	ae            *env.AppEnv
+
+	helloOutQueue chan *routerTx
+	helloInQueue  chan *routerTx
+
+	stopHelloOut chan struct{}
```

Comment:
you can probably use a single channel here if you use the close(closeNotify) pattern since they'll all see the close